packages/node_modules/@webex/internal-plugin-mercury/test/unit/spec/mercury.js

```diff
@@ -504,20 +504,54 @@ describe('plugin-mercury', () => {
       });
 
       describe('when there is a connection attempt inflight', () => {
-        it('stops the attempt', () => {
+        it('stops the attempt when disconnect called', () => {
           socketOpenStub.restore();
           socketOpenStub = sinon.stub(Socket.prototype, 'open');
-          socketOpenStub.returns(Promise.reject(new ConnectionError()));
+          socketOpenStub.onCall(0).returns(
+            // Delay the opening of the socket so that disconnect is called while open
+            // is in progress
+            promiseTick(2 * webex.internal.mercury.config.backoffTimeReset)
+            // Pretend the socket opened successfully. Failing should be fine too but
+            // it generates more console output.
+              .then(() => Promise.resolve())
+          );
           const promise = mercury.connect();
 
-          return promiseTick(1)
-            .then(() => clock.tick(webex.internal.mercury.config.backoffTimeReset))
+          // Wait for the connect call to setup
+          return promiseTick(webex.internal.mercury.config.backoffTimeReset)
             .then(() => {
+              // By this time backoffCall and mercury socket should be defined by the
+              // 'connect' call
+              assert.isDefined(mercury.backoffCall, 'Mercury backoffCall is not defined');
+              assert.isDefined(mercury.socket, 'Mercury socket is not defined');
+              // Calling disconnect will abort the backoffCall, close the socket, and
+              // reject the connect
               mercury.disconnect();
+              assert.isUndefined(mercury.backoffCall, 'Mercury backoffCall is still defined');
+              // The socket will never be unset (which seems bad)
+              assert.isDefined(mercury.socket, 'Mercury socket is not defined');
```

Comment:
Assumed the `socket` would have been blanked but that is not the case. Slightly worrying for re-connect / multiple connect scenarios.

Reply:
In the case of this test, the socket remains active after the connection is no longer valid? Just wanting to confirm based on the implemented flow.

Reply:
The socket will be closed and inactive. If another call is made to `connect()` it'll set the `socket` variable to a new instance of `Socket`. Initially I assumed the variable would be set to `undefined` to match the start state. i.e. In case the code used the existence of `socket` for any logic.